cg/meta/workflow/nf_analysis.py

```diff
@@ -631,11 +631,10 @@ def get_workflow_metrics(self, metric_id: str) -> dict:
     def get_multiqc_search_patterns(self, case_id: str) -> dict:
         """Return search patterns for MultiQC. Each key is a search pattern and each value
         corresponds to the metric ID to set in the metrics deliverables file.
-        Multiple search patterns can be added. Ideally patterns used should be sample ids, e.g.
+        Multiple search patterns can be added. Ideally, used patterns should be sample ids, e.g.
         {sample_id_1: sample_id_1, sample_id_2: sample_id_2}."""
-        sample_ids: Iterator[str] = self.status_db.get_sample_ids_by_case_id(case_id=case_id)
-        search_patterns: dict[str, str] = {sample_id: sample_id for sample_id in sample_ids}
-        return search_patterns
+        sample_ids: Iterator[str] = self.status_db.get_sample_ids_by_case_id(case_id)
+        return {sample_id: sample_id for sample_id in sample_ids}
```

Comment:
I get confused by this. Why do we have a long doc string describing something much more complex than taking all the sample ids from a case and creating a dictionary with the same keys and values from them? Also, why do we need a dictionary here if it simply maps a sample id to a sample id?

Reply:
I had the same doubts, but it is likely that this will be overrriden in some nf analysis apis to have different key, value pairs

Reply:
The docstring could be simplified though, but this is the analysis apis, it is obscure territory